strawberry/experimental/pydantic/utils.py

```diff
@@ -76,14 +76,21 @@ def has_default(model_field: DataclassCreationFields) -> bool:
     return sorted(fields, key=has_default)
 
 
-def get_default_factory_for_field(field: ModelField) -> Union[NoArgAnyCallable, _Unset]:
+def get_default_factory_for_field(
+    field: ModelField, with_input_metadata: bool = False
+) -> Union[NoArgAnyCallable, _Unset]:
     """
     Gets the default factory for a pydantic field.
 
     Handles mutable defaults when making the dataclass by using pydantic's smart_deepcopy
 
     Returns optionally a NoArgAnyCallable representing a default_factory parameter
     """
+    # Short circuit behaviour to allow usage of strawberry input parsing
+    # without losing metadata on provided payload.
+    if not field.required and with_input_metadata:
+        return lambda: UNSET
+
```

Comment:
i think theres a way where we don't need this behaviour in get_default_factory_for_field. let me try.

Reply:
Indeed. It can be inlined in type annotation and to_pydantic. I can take a swing of that as well(or another implementation) if you would provide a bit more context ;o)

Btw, I was wondering if there is an RFC or other type of document providing some background on the current concept of strawberry/pydantic interop. Would be interesting to learn more about what are the conceptual edges and how was the current implementation born, because I have a feeling that instead of interfacing with, strawberry is trying to *be* pydantic, which, to me, was suprising(and I would have said undesired), but I expect there are concerns I am missing out of lack of knowledge.

Reply:
```
def test_default_strawberry_type_with_input_metadata():
    class User2(pydantic.BaseModel):
        friend: Optional[str] = pydantic.Field(default_factory=lambda: "friend_value")
        frenemy: str

    @strawberry.experimental.pydantic.type(User2, with_input_metadata=True)
    class UserType2:
        friend: strawberry.auto
        frenemy: strawberry.auto

    user_type_2 = UserType2(frenemy="asd")
    assert user_type_2.friend == "friend_value"
```

so if you run this, the test will fail. because of the short-circuit for `get_default_factory_for_field` :( 
you'll see that in `test_default_and_default_factory`.
 the default from pydantic should be retrieved.  (well, i'm not the maintainer of this repo, so thats my opnion)

so instead of the short-circuit in `get_default_factory_for_field`, maybe retrieve the unset fields in a different function instead?

regarding the RFC, i didn't see any for it, maybe @patrick91 can comment

Reply:
> regarding the RFC, i didn't see any for it, maybe @patrick91 can comment

Unfortunately there's no RFC, the Pydantic integration has always been an experiment :D it definitely needs some polishing

I wonder if this is something we can discuss at the next strawberry monthly meeting, I don't use Pydantic a lot at the moment so I do lack knowledge around it unfortunately